Source/Common/singleton.cpp

```diff
@@ -65,8 +66,15 @@ void init_http_singleton()
             &httpSingleton,
             newSingleton
         );
+
+        if (newSingleton == nullptr)
```

Comment:
make_shared expects the allocator to throw on alloc failure, so you need your macro

Reply:
Got it